app/org/sagebionetworks/bridge/dynamodb/DynamoScheduledActivityDao.java

```diff
@@ -113,6 +129,126 @@ final void setDdbMapper(DynamoDBMapper mapper) {
     
     /** {@inheritDoc} */
     @Override
+    public ForwardCursorPagedResourceList<ScheduledActivity> getActivityHistoryV3(final String healthCode,
+            final ActivityType activityType, final String referentGuid, final DateTime scheduledOnStart,
+            final DateTime scheduledOnEnd, final String offsetKey, final int pageSize) {
+        checkNotNull(healthCode);
+        checkNotNull(activityType);
+        checkNotNull(referentGuid);
+        checkNotNull(scheduledOnStart);
+        checkNotNull(scheduledOnEnd);
+
+        int pageSizeWithIndicator = pageSize+1;
+        String start = createReferentGuid(activityType, referentGuid, scheduledOnStart.toLocalDateTime().toString());
```

Comment:
referentGuid here takes on two meanings. Could we call the param activityGuid and call type:guid:timestamp referentGuid?